build-support/bin/generate_docs.py

```diff
@@ -0,0 +1,301 @@
+            scope_to_help_info[goal][
+                "comma_separated_consumed_scopes"
+            ] = comma_separated_consumed_scopes
+
+        # Process the option data.
+
+        def munge_option(option_data):
+            # Munge the default str so we can display it nicely when it's multiline, while
+            # still displaying it inline if it's not.
+            default_str = option_data["default_str"]
+            escaped_default_str = html.escape(default_str, quote=False)
+            if "\n" in default_str:
+                option_data["marked_up_default"] = f"<pre>{escaped_default_str}</pre>"
+            else:
+                option_data["marked_up_default"] = f"<code>{escaped_default_str}</code>"
+
+        for shi in scope_to_help_info.values():
+            for opt in shi["basic"]:
+                munge_option(opt)
+            for opt in shi["advanced"]:
+                munge_option(opt)
+
+        return help_info
+
+    @property
+    def category_id(self):
+        """The id of the "Reference" category on the docsite."""
+        if self._category_id is None:
+            self._category_id = self._get_id("categories/reference")
+        return self._category_id
+
+    def _render_body(self, scope_help_info):
+        """Renders the body of a single options help page."""
+        return self._renderer.render("{{> scoped_options}}", scope_help_info)
+
+    def _access_readme_api(self, url_suffix: str, method: str, payload: str) -> Dict:
+        """Sends requests to the readme.io API."""
+        url = f"https://dash.readme.io/api/v1/{url_suffix}"
+        version = "v2.0"  # TODO: Don't hardcode this.
+
+        headers = {"content-type": "application/json", "x-readme-version": version}
+        response = requests.request(
+            method, url, data=payload, headers=headers, auth=(self._args.api_key, "")
+        )
+        response.raise_for_status()
+        return cast(Dict, response.json()) if response.text else {}
+
+    def _create(self, parent_doc_id, slug, title, body):
+        """Create a new docsite reference page.
+
+        Operates by creating a placeholder page, and then populating it via _update().
+
+        This works around a quirk of the readme.io API: You cannot set the page slug when you
+        create a page. Instead it is derived from the title.
+        In fact there is no way to set or modify the slug via the API at all, which makes sense
+        since the API references the page via the slug.  When you change the slug in the UI
+        it is likely deleting and recreating the page under the covers.
+
+        This is a problem if you want the slug to be different than the human-readable title,
+        as we do in this case. Specifically, we want the human-readable page title to be just
+        the scope name, e.g., `test` (so it appears that way in the sidebar). But we want the
+        slug to be `test-reference`, so that it doesn't collide with any other, non-generated page
```

Comment:
I recommend inverting this. We've generally been namespacing by putting the ns in the front, like `target-api-concepts` vs. `target-api-new-fields`.

This is majorly breaking to change down the road, so this is worth fixing now if we're going to change.